src/ngraph/op/topk.cpp

```diff
@@ -137,3 +137,189 @@ void op::TopK::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector&
+}
+
+size_t op::v1::TopK::read_k_from_constant_node(const shared_ptr<Node>& node,
+                                               const element::Type& k_element_type) const
+{
+    NODE_VALIDATION_CHECK(this,
+                          k_element_type == element::i8 || k_element_type == element::i32 ||
+                              k_element_type == element::i64,
+                          "K input element type must be i8, i32 or i64 (got ",
+                          k_element_type,
+                          ").");
+
+    const auto k_constant = dynamic_pointer_cast<op::Constant>(node);
+
+    size_t k = 0;
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wswitch-enum"
+#endif
+    switch (static_cast<element::Type_t>(k_element_type))
+    {
+    case element::Type_t::i8: k = validate_and_get_k<int8_t>(k_constant); break;
+    case element::Type_t::i32: k = validate_and_get_k<int32_t>(k_constant); break;
+    case element::Type_t::i64: k = validate_and_get_k<int64_t>(k_constant); break;
+    default: break;
+    }
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
+
+    return k;
+}
+
+template <typename T>
+size_t op::v1::TopK::validate_and_get_k(const shared_ptr<op::Constant>& k_constant) const
+{
+    const auto k_const_contents = k_constant->get_vector<T>();
+
+    NODE_VALIDATION_CHECK(this,
+                          k_const_contents.size() == 1,
+                          "Only one value (scalar) should be provided as the 'K' input to TopK",
+                          " (got ",
+                          k_const_contents.size(),
+                          " elements).");
+
+    NODE_VALIDATION_CHECK(this,
+                          k_const_contents[0] > 0,
+                          "The value of 'K' must be a positive number.",
+                          " (got ",
+                          k_const_contents[0],
+                          ").");
+
+    return static_cast<size_t>(k_const_contents[0]);
+}
+
+void op::v1::TopK::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    throw ngraph_error("Forward-propagation-only operation");
+}
+
+shared_ptr<Node> op::v1::TopK::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    auto new_v1_topk =
+        make_shared<v1::TopK>(new_args.at(0), new_args.at(1), m_axis, m_mode, m_sort);
+
+    new_v1_topk->set_index_element_type(m_index_element_type);
+
+    return new_v1_topk;
+}
+
+op::v1::TopK::Mode op::v1::TopK::mode_from_string(const std::string& mode) const
+{
+    static const std::map<std::string, Mode> allowed_values = {{"max", Mode::MAX},
+                                                               {"min", Mode::MIN}};
+
+    NODE_VALIDATION_CHECK(this, allowed_values.count(mode) > 0, "Invalid 'mode' value passed in.");
+
+    return allowed_values.at(mode);
+}
+
+op::v1::TopK::SortType op::v1::TopK::sort_from_string(const std::string& sort) const
```

Comment:
```suggestion
op::v1::TopK::SortType op::v1::TopK::sort_type_from_string(const std::string& sort) const
```

(Don't have strong feelings about it, just a suggestion :))